scanner/indexer/indexer.go

```diff
@@ -83,7 +84,17 @@ func NewIndexer(ctx context.Context, cfg config.IndexerConfig) (Indexer, error)
 		ScanLockRetry:        libindex.DefaultScanLockRetry,
 		LayerScanConcurrency: libindex.DefaultLayerScanConcurrency,
 	}
-
+	for name, node := range cfg.Repo {
```

Comment:
Not sure how to create a unit test for this in NewIndexer.
Any suggestions?

Reply:
Two options:

1. Refactor new into parts that can be mocked (i.e., overwrite the function name in the test, make it pieces into an object, and replace the properties/functions in the tests).
2. Put the logic you want to test in a separate function, create a unit test for it, and call from the existing `NewIndexer`.

Reply:
Worth adding a test that allows you to verify `NewIndexer` will "respect" the specs of the passed config argument.